# **C3 Project Data Cleaning**

## Objectives

* Notebook objective: Clean raw data from Kaggle

## Inputs

* BankChurners.csv file downloaded from [Kaggle](https://www.kaggle.com/datasets/sakshigoyal7/credit-card-customers) as .zip archive. 

## Process

* Unzip archive file from Kaggle to give BankChurners.csv file.  Place csv file in two directories, clean_data and raw_data.  The raw_data file will remain in the state it was after unzipping the archive file.

* The clean_data file starts out the same as the raw_data file but is processed in a sequence of steps to generate the final cleaned file.  The steps are outlined below in the notebook.

## Outputs

* At the end of the notebook a cleaned data file, BankChurners_clean.csv, is generated.

## Additional Comments

* If you have any additional comments that don't fit in the previous bullets, please state them here. 



---

## C3 Data Cleaning Process

### Data Setup

In [1]:
# Setup
import pandas as pd

df = pd.read_csv('../data/clean_data/BankChurners_clean.csv')
print(df.shape)
df.head()

(10127, 23)


,CLIENTNUM,Attrition_Flag,Customer_Age,Gender,Dependent_count,Education_Level,Marital_Status,Income_Category,Card_Category,Months_on_book,...,Credit_Limit,Total_Revolving_Bal,Avg_Open_To_Buy,Total_Amt_Chng_Q4_Q1,Total_Trans_Amt,Total_Trans_Ct,Total_Ct_Chng_Q4_Q1,Avg_Utilization_Ratio,Naive_Bayes_Classifier_Attrition_Flag_Card_Category_Contacts_Count_12_mon_Dependent_count_Education_Level_Months_Inactive_12_mon_1,Naive_Bayes_Classifier_Attrition_Flag_Card_Category_Contacts_Count_12_mon_Dependent_count_Education_Level_Months_Inactive_12_mon_2
0,768805383,Existing Customer,45,M,3,High School,Married,$60K - $80K,Blue,39,...,12691.0,777,11914.0,1.335,1144,42,1.625,0.061,0.000093,0.99991
1,818770008,Existing Customer,49,F,5,Graduate,Single,Less than $40K,Blue,44,...,8256.0,864,7392.0,1.541,1291,33,3.714,0.105,0.000057,0.99994
2,713982108,Existing Customer,51,M,3,Graduate,Married,$80K - $120K,Blue,36,...,3418.0,0,3418.0,2.594,1887,20,2.333,0.000,0.000021,0.99998
3,769911858,Existing Customer,40,F,4,High School,Unknown,Less than $40K,Blue,34,...,3313.0,2517,796.0,1.405,1171,20,2.333,0.760,0.000134,0.99987
4,709106358,Existing Customer,40,M,3,Uneducated,Married,$60K - $80K,Blue,21,...,4716.0,0,4716.0,2.175,816,28,2.500,0.000,0.000022,0.99998


In [6]:
# Lesson: Use df.info(), df.describe(), and df.isna().sum() to map the damage.
# Try it:
#   Check the column names. Notice any spaces or weird capitalisation?
#   Print df.isna().sum() to see which columns have missing values.

# --- YOUR CODE ---
print(df.info())
print('=============================')
print(df.describe())
print('=============================')
df.isna().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10127 entries, 0 to 10126
Data columns (total 23 columns):
 #   Column                                                                                                                              Non-Null Count  Dtype  
---  ------                                                                                                                              --------------  -----  
 0   CLIENTNUM                                                                                                                           10127 non-null  int64  
 1   Attrition_Flag                                                                                                                      10127 non-null  object 
 2   Customer_Age                                                                                                                        10127 non-null  int64  
 3   Gender                                                                           

CLIENTNUM                                                                                                                             0
Attrition_Flag                                                                                                                        0
Customer_Age                                                                                                                          0
Gender                                                                                                                                0
Dependent_count                                                                                                                       0
Education_Level                                                                                                                       0
Marital_Status                                                                                                                        0
Income_Category                                 

The output confirms that the file contains 10,127 records.  Inspection of the output reveals that all variable names, except for CLIENTNUM, are written in CapWords, also known as PascalCase.  There are no obvious misspellings of any of the variable names.  